invoice_modify_header/invoice.py

```diff
@@ -0,0 +1,125 @@
+from trytond.pool import Pool, PoolMeta
+from trytond.model import Model, ModelView
+from trytond.pyson import Eval
+from trytond.wizard import Button, StateTransition, StateView, Wizard
+from trytond.model.exceptions import AccessError
+from trytond.i18n import gettext
+from trytond.transaction import Transaction
+
+
+class Invoice(metaclass=PoolMeta):
+    __name__ = 'account.invoice'
+
+    @classmethod
+    def __setup__(cls):
+        super().__setup__()
+        # readonly party in case has lines
+        _READONLY = (cls.party.states['readonly']
+            | (Eval('lines', [0]) & Eval('party')))
+        cls.party.states['readonly'] = _READONLY
+
+        cls._buttons.update({
+                'modify_header': {
```

Comment:
I'm not sure if it will be better to do an action, no a button.

Reply:
Same UI that sale and purchase modules